tests/TestCase/Controller/SentencesListsControllerTest.php

```diff
@@ -41,7 +41,7 @@ public function accessesProvider() {
             [ '/en/sentences_lists/show/', null, '/en/sentences_lists/index' ],
             [ '/en/sentences_lists/show/1', null, true ],
             [ '/en/sentences_lists/show/1', 'kazuki', true ],
-            [ '/en/sentences_lists/show/1/fra', 'kazuki', true ],
+            [ '/en/sentences_lists/show/1/und/fra', 'kazuki', true ],
```

Comment:
By the way, feel free to add a new unit test testing for the new redirection code.